File: 5/solution.py
class Solution:
    def find(self, i: int, j: int):
        while (i >= 0 and j < self.n):
            if self.s[i] != self.s[j]:
                break
            i -= 1
            j += 1
        if j - i - 1 > self.length:
            self.length = j - i - 1
            self.begin = i + 1
            self.end = j - 1

    def longestPalindrome(self, s: str) -> str:
        self.s = s
        self.n = len(s)
        self.length = 1
        self.begin = 0
        self.end = 0
        for i in range(self.n):
            self.find(i, i)
            self.find(i, i + 1)
        return s[self.begin:self.end + 1]

File: 5/solution.go
package _5

func longestPalindrome(s string) string {
	n, length := len(s), 1
	begin, end := 0, 0
	find := func(i, j int) {
		for i >= 0 && j < n {
			if s[i] != s[j] {
				break
			}
			i--
			j++
		}
		if j-i-1 > length {
			length = j - i - 1
			begin = i + 1
			end = j - 1
		}
	}
	for i := 0; i < n; i++ {
		find(i, i)
		find(i, i+1)
	}
	return s[begin : end+1]
}
